### PR TITLE
Add Mamba and Mathstral to README

### PR DESCRIPTION
**Pull request by patrickvonplaten**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -150,6 +154,36 @@ This function uses recursion to calculate the Fibonacci number. However, it's no
...
+pip install packaging mamba-ssm causal-conv1d transformers
+```
+
+before you can start chatting:
+
+```sh
+mistral-chat $7B_MAMBA_CODESTRAL --instruct --max_tokens 256
+```
+
+- **Chat as Math Assistant**
```
Comment (timlacroix) [inline README.md:174]:
```suggestion
- **Chat with Mathstral**
```

File: README.md [hunk 2]
```diff
@@ -150,6 +154,36 @@ This function uses recursion to calculate the Fibonacci number. However, it's no
...
+To use [Mathstral](https://mistral.ai/news/mathstral/) as an assistant you can run the following command using `mistral-chat`.
+Make sure `$M7B_MATHSTRAL` is set to a valid path to the downloaded codestral folder, e.g. `$HOME/mistral_models/mathstral-7B-v0.1`
+
+```sh
+mistral-chat $M7B_MATHSTRAL --instruct --max_tokens 256
+```
+
+If you prompt it with *"Albert likes to surf every week. Each surfing session lasts for 4 hours and costs $20 per hour. How much world Albert spend in 5 weeks?"*, the model should answer with the correct calculation.
+
+You can then continue chatting afterwards, *e.g.* with *"How much would he spent in a year?"*.
```
Comment (timlacroix) [inline README.md:185]:
```suggestion
You can then continue chatting afterwards, *e.g.* with *"How much would he spend in a year?"*.
```

File: README.md [hunk 3]
```diff
@@ -150,6 +154,36 @@ This function uses recursion to calculate the Fibonacci number. However, it's no
...
+- **Chat with Mathstral**
+
+To use [Mathstral](https://mistral.ai/news/mathstral/) as an assistant you can run the following command using `mistral-chat`.
+Make sure `$M7B_MATHSTRAL` is set to a valid path to the downloaded codestral folder, e.g. `$HOME/mistral_models/mathstral-7B-v0.1`
+
+```sh
+mistral-chat $M7B_MATHSTRAL --instruct --max_tokens 256
+```
+
+If you prompt it with *"Albert likes to surf every week. Each surfing session lasts for 4 hours and costs $20 per hour. How much world Albert spend in 5 weeks?"*, the model should answer with the correct calculation.
```
Comment (timlacroix) [inline README.md:183]:
```suggestion
If you prompt it with *"Albert likes to surf every week. Each surfing session lasts for 4 hours and costs $20 per hour. How much would Albert spend in 5 weeks?"*, the model should answer with the correct calculation.
```

File: README.md [hunk 4]
```diff
@@ -150,6 +154,36 @@ This function uses recursion to calculate the Fibonacci number. However, it's no
 
 You can continue chatting afterwards, *e.g.* with *"Translate it to Python"*.
 
+- **Chat with Mamba-Codestral**
+
+To use [Mamba-Codestral](https://mistral.ai/news/codestral/) as a coding assistant you can run the following command using `mistral-chat`.
+Make sure `$7B_MAMBA_CODESTRAL` is set to a valid path to the downloaded codestral folder, e.g. `$HOME/mistral_models/mamba-codestral-7B-v0.1`.
+
+You then need to addiitonally install the following packages:
```
Comment (timlacroix) [inline README.md:162]:
```suggestion
You then need to additionally install the following packages:
```

File: README.md [hunk 5]
```diff
@@ -118,7 +122,7 @@ torchrun --nproc-per-node 2 --no-python mistral-chat $M8x7B_DIR --instruct
 
 *Note*: Change `--nproc-per-node` to more GPUs if necessary (*e.g.* for 8x22B).
 
-- **Chat as Code Assistant**
+- **Chat with Codestral**
 
 To use [Codestral](https://mistral.ai/news/codestral/) as a coding assistant you can run the following command using `mistral-chat`.
```
Comment (pandora-s-git) [inline README.md:127]: https://mistral.ai/news/codestral-mamba/ ?

File: README.md [hunk 6]
```diff
@@ -150,6 +154,36 @@ This function uses recursion to calculate the Fibonacci number. However, it's no
 
 You can continue chatting afterwards, *e.g.* with *"Translate it to Python"*.
 
+- **Chat with Mamba-Codestral**
+
+To use [Mamba-Codestral](https://mistral.ai/news/codestral/) as a coding assistant you can run the following command using `mistral-chat`.
+Make sure `$7B_MAMBA_CODESTRAL` is set to a valid path to the downloaded codestral folder, e.g. `$HOME/mistral_models/mamba-codestral-7B-v0.1`.
```
Comment (timlacroix) [inline README.md:160]:
```suggestion
Make sure `$7B_MAMBA_CODESTRAL` is set to a valid path to the downloaded codestral-mamba folder, e.g. `$HOME/mistral_models/mamba-codestral-7B-v0.1`.
```
```suggestion
Make sure `$7B_MAMBA_CODESTRAL` is set to a valid path to the downloaded codestral folder, e.g. `$HOME/mistral_models/mamba-codestral-7B-v0.1`.
```

File: README.md [hunk 7]
```diff
@@ -10,6 +10,8 @@ Blog 7B: [https://mistral.ai/news/announcing-mistral-7b/](https://mistral.ai/new
 Blog 8x7B: [https://mistral.ai/news/mixtral-of-experts/](https://mistral.ai/news/mixtral-of-experts/)\
 Blog 8x22B: [https://mistral.ai/news/mixtral-8x22b/](https://mistral.ai/news/mixtral-8x22b/)\
 Blog Codestral 22B: [https://mistral.ai/news/codestral](https://mistral.ai/news/codestral/)
+Blog Mamba Codestral 7B: [https://mistral.ai/news/codestral-mamba/](https://mistral.ai/news/codestral-mamba/)
```
Comment (timlacroix) [inline README.md:13]:
```suggestion
Blog Codestral Mamba 7B: [https://mistral.ai/news/codestral-mamba/](https://mistral.ai/news/codestral-mamba/)
```

File: README.md [hunk 8]
```diff
@@ -150,6 +154,36 @@ This function uses recursion to calculate the Fibonacci number. However, it's no
 
 You can continue chatting afterwards, *e.g.* with *"Translate it to Python"*.
 
+- **Chat with Mamba-Codestral**
+
+To use [Mamba-Codestral](https://mistral.ai/news/codestral/) as a coding assistant you can run the following command using `mistral-chat`.
```
Comment (patrickvonplaten) [inline README.md:159]:
```suggestion
To use [Mamba-Codestral](https://mistral.ai/news/codestral-mamba/) as a coding assistant you can run the following command using `mistral-chat`.
```

File: README.md [hunk 9]
```diff
@@ -43,6 +45,8 @@ cd $HOME/mistral-inference && poetry install .
 | 8x7B |     **Updated model coming soon!**       | - |
 | 8x22B | https://models.mistralcdn.com/mixtral-8x22b-v0-3/mixtral-8x22B-v0.3.tar | `a2fa75117174f87d1197e3a4eb50371a` |
 | Codestral 22B | https://models.mistralcdn.com/codestral-22b-v0-1/codestral-22B-v0.1.tar | `1ea95d474a1d374b1d1b20a8e0159de3` |
+| Mathstral 7B | https://models.mistralcdn.com/mathstral-7b-v0-1/mathstral-7B-v0.1.tar | `5f05443e94489c261462794b1016f10b` |
+| Mamba-Codestral 7B | https://models.mistralcdn.com/codestral-mamba-7b-v0-1/codestral-mamba-7B-v0.1.tar | `d3993e4024d1395910c55db0d11db163` |
```
Comment (timlacroix) [inline README.md:49]:
```suggestion
| Codestral-Mamba 7B | https://models.mistralcdn.com/codestral-mamba-7b-v0-1/codestral-mamba-7B-v0.1.tar | `d3993e4024d1395910c55db0d11db163` |
```

File: README.md [hunk 10]
```diff
@@ -150,6 +154,36 @@ This function uses recursion to calculate the Fibonacci number. However, it's no
 
 You can continue chatting afterwards, *e.g.* with *"Translate it to Python"*.
 
+- **Chat with Mamba-Codestral**
```
Comment (timlacroix) [inline README.md:157]:
```suggestion
- **Chat with Codestral-Mamba**
```

File: README.md (hunk 5 above)
Comment (patrickvonplaten) [reply README.md:127]: Should be in the section below :-)

File: README.md [hunk 11]
```diff
@@ -150,6 +154,36 @@ This function uses recursion to calculate the Fibonacci number. However, it's no
 
 You can continue chatting afterwards, *e.g.* with *"Translate it to Python"*.
 
+- **Chat with Codestral-Mamba**
+
+To use [Mamba-Codestral](https://mistral.ai/news/codestral-mamba/) as a coding assistant you can run the following command using `mistral-chat`.
```
Comment (timlacroix) [inline README.md:159]:
```suggestion
To use [Codestral-Mamba](https://mistral.ai/news/codestral-mamba/) as a coding assistant you can run the following command using `mistral-chat`.
```

File: README.md [hunk 12]
```diff
@@ -150,6 +154,36 @@ This function uses recursion to calculate the Fibonacci number. However, it's no
 
 You can continue chatting afterwards, *e.g.* with *"Translate it to Python"*.
 
+- **Chat with Codestral-Mamba**
+
+To use [Codestral-Mamba](https://mistral.ai/news/codestral-mamba/) as a coding assistant you can run the following command using `mistral-chat`.
+Make sure `$7B_MAMBA_CODESTRAL` is set to a valid path to the downloaded codestral-mamba folder, e.g. `$HOME/mistral_models/mamba-codestral-7B-v0.1`.
```
Comment (timlacroix) [inline README.md:160]:
```suggestion
Make sure `$7B_CODESTRAL_MAMBA` is set to a valid path to the downloaded codestral-mamba folder, e.g. `$HOME/mistral_models/mamba-codestral-7B-v0.1`.
```

File: README.md [hunk 13]
```diff
@@ -150,6 +154,36 @@ This function uses recursion to calculate the Fibonacci number. However, it's no
...
+You then need to additionally install the following packages:
+  
+```
+pip install packaging mamba-ssm causal-conv1d transformers
+```
+
+before you can start chatting:
+
+```sh
+mistral-chat $7B_MAMBA_CODESTRAL --instruct --max_tokens 256
```
Comment (timlacroix) [inline README.md:171]:
```suggestion
mistral-chat $7B_CODESTRAL_MAMBA --instruct --max_tokens 256
```

File: README.md [hunk 14]
```diff
@@ -150,6 +154,36 @@ This function uses recursion to calculate the Fibonacci number. However, it's no
...
+before you can start chatting:
+
+```sh
+mistral-chat $7B_CODESTRAL_MAMBA --instruct --max_tokens 256
+```
+
+- **Chat with Mathstral**
+
+To use [Mathstral](https://mistral.ai/news/mathstral/) as an assistant you can run the following command using `mistral-chat`.
+Make sure `$M7B_MATHSTRAL` is set to a valid path to the downloaded codestral folder, e.g. `$HOME/mistral_models/mathstral-7B-v0.1`
```
Comment (timlacroix) [inline README.md:177]:
```suggestion
Make sure `$7B_MATHSTRAL` is set to a valid path to the downloaded codestral folder, e.g. `$HOME/mistral_models/mathstral-7B-v0.1`
```

File: README.md [hunk 15]
```diff
@@ -150,6 +154,36 @@ This function uses recursion to calculate the Fibonacci number. However, it's no
...
+mistral-chat $7B_CODESTRAL_MAMBA --instruct --max_tokens 256
+```
+
+- **Chat with Mathstral**
+
+To use [Mathstral](https://mistral.ai/news/mathstral/) as an assistant you can run the following command using `mistral-chat`.
+Make sure `$7B_MATHSTRAL` is set to a valid path to the downloaded codestral folder, e.g. `$HOME/mistral_models/mathstral-7B-v0.1`
+
+```sh
+mistral-chat $M7B_MATHSTRAL --instruct --max_tokens 256
```
Comment (timlacroix) [inline README.md:180]:
```suggestion
mistral-chat $7B_MATHSTRAL --instruct --max_tokens 256
```

**Review by timlacroix — APPROVED** [review]
all good thx !

**Pull request merged**